spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/annotation/SqsListener.java

```diff
@@ -0,0 +1,131 @@
+ * {@link io.awspring.cloud.sqs.listener.AsyncMessageListener} and set to a
+ * {@link io.awspring.cloud.sqs.listener.MessageListenerContainer}.
+ * <p>
+ * Each method will be handled by a different container instance, created by the specified {@link #factory()} property.
+ * If not specified, a default factory will be looked up in the context.
+ * <p>
+ * When used in conjunction with Spring Boot and auto configuration, the framework supplies a default
+ * {@link io.awspring.cloud.sqs.config.SqsMessageListenerContainerFactory} and a
+ * {@link software.amazon.awssdk.services.sqs.SqsAsyncClient}, unless such beans are already found in the
+ * {@link org.springframework.context.ApplicationContext}.
+ * <p>
+ * For more complex configurations, {@link io.awspring.cloud.sqs.config.SqsMessageListenerContainerFactory} instances
+ * can be created and configured. See {@link SqsMessageListenerContainerFactory#builder()} for more information on
+ * creating and configuring a factory.
+ * <p>
+ * Further configuration for containers created using this annotation can be achieved by declaring
+ * {@link io.awspring.cloud.sqs.config.SqsListenerCustomizer} beans.
+ * <p>
+ * Methods with this annotation can have flexible signatures, including arguments of the following types:
+ * <ul>
+ * <li>{@link org.springframework.messaging.handler.annotation.Header}</li>
+ * <li>{@link org.springframework.messaging.handler.annotation.Headers}</li>
+ * <li>{@link org.springframework.messaging.Message}</li>
+ * <li>{@link io.awspring.cloud.sqs.listener.Visibility}</li>
+ * <li>{@link io.awspring.cloud.sqs.listener.QueueAttributes}</li>
+ * <li>{@link software.amazon.awssdk.services.sqs.model.Message}</li>
+ * <li>{@link io.awspring.cloud.sqs.listener.acknowledgement.Acknowledgement}</li>
+ * <li>{@link io.awspring.cloud.sqs.listener.acknowledgement.AsyncAcknowledgement}</li>
+ * </ul>
+ * Method signatures also accept {@link java.util.List}&lt;Pojo&gt; and
+ * {@link java.util.List}{@link org.springframework.messaging.Message}&lt;Pojo&gt; arguments . Such arguments will
+ * configure the container to batch mode. When using List arguments, no other arguments can be provided. Metadata can be
+ * retrieved by inspecting the {@link org.springframework.messaging.Message} instances'
+ * {@link org.springframework.messaging.MessageHeaders}.
+ * <p>
+ * To support {@link io.awspring.cloud.sqs.listener.acknowledgement.Acknowledgement} and
+ * {@link io.awspring.cloud.sqs.listener.acknowledgement.AsyncAcknowledgement} arguments, the factory used to create the
+ * containers must be set to {@link io.awspring.cloud.sqs.listener.acknowledgement.handler.AcknowledgementMode#MANUAL}.
+ * <p>
+ * Properties in this annotation support property placeholders ("${...}") and SpEL ("#{...}").
+ *
+ * @see SqsMessageListenerContainerFactory
+ * @see io.awspring.cloud.sqs.listener.SqsMessageListenerContainer
+ * @see io.awspring.cloud.sqs.config.SqsListenerCustomizer
```

Comment:
IMO instead of having the FQN those can be imported and just have `{@link Header}` or `@see SqsMessageListernerContainer`

Reply:
Yeah, TBH this is kind of a gray area for me, which concerns package tangles.

AFAIK, if we add the imports, dependency analyzers will consider that a dependency and increases circular dependencies in packages.

That being said, I don't really know what the fuzz is about when it comes to not having dependencies from packages to each other.

Of course, I get that ideally dependencies would flow only in one direction, but in practice what I see is moving lots of classes to the same package and having bloated packages to avoid these dependencies.

Or, in this case, having `FQCN` instead of just the class name.

So I'm cool with adding the imports, just want to make sure package tangles are not a problem we'll need to solve later.